src/main/java/lembas/core/api/common/SimpleLogger.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (C) 2020 Lembas Modding Team
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package lembas.core.api.common;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * A simple wrapper for a logger using String.format for formatting the log
+ * messages. The [...]Throwable functions are there to avoid bugs related to the

```

Comment:
How does that help?

Reply:
The problem with the normal logger interfaces (Log4j etc.) is, that they only have varargs, and for example the first one, if it's an exception, will be threated as such one. Now if I, by accident, swap two arguments, the exception will be threated as a normal parameter and may, depending on the message formatting, be included in the message (with a completely wrong meaning then, because it's at the place where the other argument should be), or just swallowed completely. This approach gives more safety, because the exception argument is explicitely needed and not "swallowed" with the varargs, so if the programmer intends that the exception instances is logged, this ensures that it really is.